storybook/stories/contributing/contributing-components.txt

```diff
@@ -0,0 +1,347 @@
+
+Step 1: [Do we need this component?](#step-1)
+Step 2: [Open a GitHub issue](#step-2) 
+Step 3: [Implement your change](#step-3) 
+Step 4: [Refinement](#step-4) 
+Step 5: [Publish and Maintain](#step-5) 
+
+#### Step 1: Do we need this component? {#step-1}
+
+It’s important to ask yourself this question before starting your contribution. Our library should include components that are generic and flexible enough to work across various products. It should include what’s shared across products and leave out what’s not.
+
+A great first step is to share your idea with our design community to see if there’s a need for your component. You can leave a comment in the [Design Systems Figma Library](https://www.figma.com/design/ficiK5ADsOuiCromUZam1E/WordPress-Design-System-(Community)?node-id=13482-72792&t=ofEJnkdwKdS5mhGX-1), or drop a question in #design-systems on Slack, where people will be happy to help.
+
+Some more questions you can ask yourself if you’re thinking of creating a new component:
+
+- Could it be used by other products or plugins?  
+- Does it overlap functionally or visually with any existing components? Can we use existing components to implement an alternative solution?  
+- How much effort will be required to create and maintain it?  
+- Is there a clear purpose for it?  
+- Can we introduce this change without causing breaking changes? Will we be able to keep this change and iterate on it in the future without causing future breaking changes?
+
+This flowchart can help determine if a new component is necessary:
+
+ADD IMAGE
+
+#### Step 2: Open a GitHub issue {#step-2}
+
+If you’ve decided you want to create a new component, you’ll need to open a GitHub issue on Project Gutenberg. This is where we’ll be able to discuss the proposed changes and track progress.
+
+[Click here to open a new issue](https://github.com/WordPress/gutenberg/issues/new?labels=Design+System&body=Then%2C+write+a+detailed+description%3A%0D%0A%0D%0A-+Explain+your+rationale+for+adding+this+component%0D%0A-+Detail+the+intended+behavior%0D%0A-+Clarify+if+it%E2%80%99s+a+new+component+or+a+variation+of+an+existing+one%0D%0A-+Include+a+basic+mockup+%28optional%29%0D%0A-+Include+inspiration+from+other+products+%28optional%29).
+
+Then, write a detailed description:
+
+- Explain your rationale for adding this component  
+- Detail the intended behavior  
+- Clarify if it’s a new component or a variation of an existing one  
+- Include a basic mockup (optional)  
+- Include inspiration from other products (optional)  
+- Tag the [@WordPress/gutenberg-components](https://github.com/orgs/WordPress/teams/gutenberg-components) and [@WordPress/gutenberg-design](https://github.com/orgs/WordPress/teams/gutenberg-design) GitHub groups
+
+Add the “Design System” label to correctly categorize your work. Also, consider adding other labels depending on your needs, like Needs Design, Needs Copy Review, Needs Dev, or Needs Accessibility Feedback.
```

Comment:
I see what you mean, but wouldn't it be helpful for a new contributor to get used to labeling their issues correctly?

Reply:
They literally [cannot](https://stackoverflow.com/a/13829505) because they do not have the rights. I believe it works the same on almost all open source projects on GitHub. They gain rights when they join the [Triage team](https://developer.wordpress.org/block-editor/contributors/triage/) or meet the criteria to join the [Gutenberg team](https://developer.wordpress.org/block-editor/contributors/repository-management/#teams).